src/Features/VisualBasic/Portable/ValidateFormatString/ValidateFormatStringDiagnosticAnalyzer.vb

```diff
@@ -0,0 +1,102 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+Imports Microsoft.CodeAnalysis.Diagnostics
+Imports Microsoft.CodeAnalysis.ValidateFormatString
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.ValidateFormatString
+    <DiagnosticAnalyzer(LanguageNames.VisualBasic)>
+    Friend Class ValidateFormatStringDiagnosticAnalyzer
+        Inherits AbstractValidateFormatStringDiagnosticAnalyzer
+
+        Public Overrides Sub Initialize(context As AnalysisContext)
+            context.RegisterSyntaxNodeAction(AddressOf AnalyzeNode, SyntaxKind.InvocationExpression)
+        End Sub
+
+        Private Sub AnalyzeNode(context As SyntaxNodeAnalysisContext)
+            Dim invocationExpr = CType(context.Node, InvocationExpressionSyntax)
```

Comment:
this logic looks quite similar to hte C# version.  Can we share more here (potentially using generics, or using ISyntaxFacts service)?